x/batch/decorator_test.go

```diff
@@ -89,190 +116,208 @@ func mockLog(num int64, content string) string {
+				Data:        mockData(4, data),
+				Log:         mockLog(4, logVal),
+				GasUsed:     gas * 4,
+				Diff:        mockDiff(4),
+				Tags:        mockTags(4),
+				RequiredFee: coin.Coin{Whole: 3, Fractional: 50, Ticker: "IOV"},
+			},
+		},
+
+		"wrong message type works fine and has no effect": {
+			msg:        &wrongWeaveMsg{},
+			deliverRes: &weave.DeliverResult{},
+			checkRes:   &weave.CheckResult{},
+		},
+
+		"tx GetMsg error": {
+			msg:   &mockMsg{},
+			txErr: errors.ErrInput,
+			err:   errors.ErrInput,
+		},
+
+		"incompatible fees": {
+			msg: &mockMsg{list: make([]weave.Msg, 2)},
+			err: errors.ErrInput,
+			check: &checkMock{
+				res: []*weave.CheckResult{
+					{
+						Data:         make([]byte, 2),
+						Log:          logVal,
+						GasAllocated: gas,
+						GasPayment:   gas,
+						RequiredFee:  coin.Coin{Whole: 1, Ticker: "IOV"},
+					},
+					{
+						Data:         make([]byte, 1),
+						Log:          logVal,
+						GasAllocated: gas,
+						GasPayment:   gas,
+						RequiredFee:  coin.Coin{Whole: 1, Ticker: "LSK"},
+					},
+				},
+			},
+		},
+		"validation error": {
+			msg: &mockMsg{list: make([]weave.Msg, 2), valErr: errors.ErrOverflow},
+			err: errors.ErrOverflow,
+		},
+
+		"error while executing one of the messages": {
+			msg: &mockMsg{list: make([]weave.Msg, 4)},
+
+			deliver: &deliverMock{
+				res: []*weave.DeliverResult{
+					makeRes(coin.Coin{Whole: 1, Fractional: 50, Ticker: "IOV"}),
+					makeRes(coin.Coin{}),
+				},
+				err: []error{
+					nil, errors.ErrType,
+				},
+			},
+			err: errors.ErrType,
+			check: &checkMock{
+				res: []*weave.CheckResult{
+					{
+						Data:         make([]byte, 2),
+						Log:          logVal,
+						GasAllocated: gas,
+						GasPayment:   gas,
+						RequiredFee:  coin.Coin{Whole: 1, Ticker: "IOV"},
+					},
+					{
+						Data:         make([]byte, 1),
+						Log:          logVal,
+						GasAllocated: gas,
+						GasPayment:   gas,
+						RequiredFee:  coin.Coin{Whole: 1, Ticker: "LSK"},
+					},
+				},
+				err: []error{
+					nil, errors.ErrType,
+				},
+			},
+		},
+	}
 
-		Convey("Combine required fees with none", func() {
-			// 4 elements, 1 and 3 with fees, 2 and 4 without
-			num := int64(4)
-			logVal := "log"
-			dataContent := make([]byte, 2)
-			gas := int64(1)
-
-			fee := coin.Coin{Whole: 1, Fractional: 50, Ticker: "IOV"}
-			fee2 := coin.Coin{Whole: 2, Ticker: "IOV"}
-			combined := coin.Coin{Whole: 3, Fractional: 50, Ticker: "IOV"}
-			zero := coin.Coin{}
-
-			msg.On("Validate").Return(nil).Times(2)
-			msg.On("MsgList").Return(make([]weave.Msg, 4), nil).Times(2)
-			helper.On("GetMsg").Return(msg, nil).Times(2)
-			makeRes := func(req coin.Coin) *weave.DeliverResult {
-				return &weave.DeliverResult{
-					Data:        dataContent,
-					Log:         logVal,
-					GasUsed:     gas,
-					Diff:        make([]weave.ValidatorUpdate, 1),
-					Tags:        make([]common.KVPair, 1),
-					RequiredFee: req,
+	for msg, spec := range specs {
```

Comment:
I think using `msg` for test name variable can be missleading. We are using `msg` variable name everywhere to reference `weave.Msg`.

Reply:
This is used across a lot of our other tests, I'd like to keep this uniform.